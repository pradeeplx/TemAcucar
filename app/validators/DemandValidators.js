export default DemandValidators = {
  radius: {
    title: 'Raio',
    validate: [{
      validator: 'isNumeric',
    }]
  },
  name: {
    title: 'O que você precisa?',
    validate: [{
      validator: 'isLength',
      arguments: [1, 255],
    }]
  },
  description: {
    title: 'Para que você vai usar?',
    validate: [{
      validator: 'isLength',
      arguments: [1, 255],
    }]
  },
  errorMessage: () => 'Oops! Ocorreu um erro ao acessar nosso servidor. Por favor, tente novamente.',
}
