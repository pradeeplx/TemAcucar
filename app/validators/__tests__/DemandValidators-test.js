import 'react-native'
import React from 'react'
import DemandValidators from '../DemandValidators'

test('DemandValidators object', () => {
  expect(DemandValidators).toMatchSnapshot()
})

describe('DemandValidators.errorMessage cases', () => {
  it('has a default errorMessage', () => {
    expect(DemandValidators.errorMessage({ id: 'foo' })).toMatchSnapshot()
  })
})