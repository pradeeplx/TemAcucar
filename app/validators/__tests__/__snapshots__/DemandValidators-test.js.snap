exports[`DemandValidators.errorMessage cases has a default errorMessage 1`] = `"Oops! Ocorreu um erro ao acessar nosso servidor. Por favor, tente novamente."`;

exports[`test DemandValidators object 1`] = `
Object {
  "description": Object {
    "title": "Para que você vai usar?",
    "validate": Array [
      Object {
        "arguments": Array [
          1,
          255,
        ],
        "validator": "isLength",
      },
    ],
  },
  "errorMessage": [Function],
  "name": Object {
    "title": "O que você precisa?",
    "validate": Array [
      Object {
        "arguments": Array [
          1,
          255,
        ],
        "validator": "isLength",
      },
    ],
  },
  "radius": Object {
    "title": "Raio",
    "validate": Array [
      Object {
        "validator": "isNumeric",
      },
    ],
  },
}
`;
